src/services/localStorageService.js

```diff
@@ -0,0 +1,34 @@
+export const removeFromHistory = placeId => {
+  const locationsHistory = JSON.parse(localStorage.getItem('locationsHistory')) || [];
+  const locationIndex = locationsHistory.findIndex(location => location.placeId === placeId);
+
+  if (locationIndex === -1) {
+    return;
+  }
+
+  locationsHistory.splice(locationIndex, 1);
+  localStorage.setItem('locationsHistory', JSON.stringify(locationsHistory));
+};
+
+export const addToHistory = ({ placeName, placeId, lat, lng }) => {
+  removeFromHistory(placeId);
+
+  const locationsHistory = JSON.parse(localStorage.getItem('locationsHistory')) || [];
+
+  if (locationsHistory.length > 4) {
+    locationsHistory.shift();
+  }
+
+  locationsHistory.push({ placeName, placeId, lat, lng });
```

Comment:
just a suggestion > `updatedHistory = [{ placeName, placeId, lat, lng }, ...locationHistory].splice(0,4)` this probably will do a same without if clause